File: solutions/kotlin/raindrops/1/src/main/kotlin/Raindrops.kt
object Raindrops {

    fun convert(n: Int): String {
	val result = StringBuilder()
	if (n.mod(3) == 0)
	    result.append("Pling")
	if (n.mod(5) == 0)
	    result.append("Plang")
	if (n.mod(7) == 0)
	    result.append("Plong")
	return if (result.isEmpty()) "$n" else result.toString()
    }
}

File: solutions/fortran/raindrops/1/raindrops.f90
module raindrops
  implicit none

  type :: raindrop
     integer :: k
     character(5) :: v
  end type raindrop
contains

  function convert(i)
    integer :: i, n, c
    character(20) :: convert
    type(raindrop) :: translation(3)
    convert = repeat(" ", 20)
    translation = [raindrop(3, "Pling"), raindrop(5, "Plang"), raindrop(7, "Plong")]
    c = 1
    do n = 1, size(translation)
       if (mod(i, translation(n)%k) == 0) then
          convert(c:) = translation(n)%v
          c = c + len(translation(n)%v)
       end if
       print *, i, "%", translation(n)%k, mod(i, translation(n)%k)
    end do
    print *, "convert = ", convert
    if (convert(1:1) == " ") then
       write(convert, "(i20)") i
       convert = trim(adjustl(convert))
    end if
  end function convert

end module raindrops
